File: Task-03/Java/Sub-Task 4/Pattern-FileHandling2.java
import java.io.IOException;
import java.nio.file.*;
import java.util.List;

public class DiamondPattern {
    public static void main(String[] args) throws IOException {
        List<String> lines = Files.readAllLines(Paths.get("input.txt"));
        int n = Integer.parseInt(lines.get(0).trim());

        StringBuilder output = new StringBuilder();
        for (int i = 0; i < n; i++) {
            output.append(" ".repeat(n - i - 1)).append("*".repeat(2 * i + 1)).append("\n");
        }
        for (int i = n - 2; i >= 0; i--) {
            output.append(" ".repeat(n - i - 1)).append("*".repeat(2 * i + 1)).append("\n");
        }

        Files.write(Paths.get("output.txt"), output.toString().getBytes());
    }
}

File: Task-03/Rust/Sub-Task 4/Pattern-FileHandling2.rs
use std::fs;

fn main() {
    let n: usize = fs::read_to_string("input.txt").unwrap().trim().parse().unwrap();

    let mut output = String::new();
    for i in 0..n {
        output.push_str(&format!("{:width$}{}\n", "", "*".repeat(2 * i + 1), width = n - i - 1));
    }
    for i in (0..n - 1).rev() {
        output.push_str(&format!("{:width$}{}\n", "", "*".repeat(2 * i + 1), width = n - i - 1));
    }

    fs::write("output.txt", output).expect("Unable to write file");
}
